ES6 syntax for import/export doesn't work in ambient modules

Code to reproduce:
```ts
// File: external.d.ts
export interface IFoo {}
```
```ts
// File: ambient.d.ts
declare module "moduleA" {
    import { IFoo } from "external"; // ERROR TS2307: Cannot find module 'external'.
}
```
Old syntax with `require` works fine:
```ts
// File: ambient.d.ts
declare module "moduleA" {
    import ext = require("external");
    import IFoo = ext.IFoo; // works
}
```
But I can't use `require` to `export *`. This is a problem (`export =` is not the same)
Tested in TS 1.5.3 and 1.6.2 with `--module amd` compiler option.

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -948,12 +948,6 @@ namespace ts {
             return symbol.flags & meaning ? symbol : resolveAlias(symbol);
         }
 
-        function isExternalModuleNameRelative(moduleName: string): boolean {
-            // TypeScript 1.0 spec (April 2014): 11.2.1
-            // An external module name is "relative" if the first term is "." or "..".
-            return moduleName.substr(0, 2) === "./" || moduleName.substr(0, 3) === "../" || moduleName.substr(0, 2) === ".\\" || moduleName.substr(0, 3) === "..\\";
-        }
-
         function resolveExternalModuleName(location: Node, moduleReferenceExpression: Expression): Symbol {
             if (moduleReferenceExpression.kind !== SyntaxKind.StringLiteral) {
                 return;

--- a/src/compiler/program.ts
+++ b/src/compiler/program.ts
@@ -692,6 +692,12 @@ namespace ts {
             
             let imports: LiteralExpression[];
             for (let node of file.statements) {
+                collect(node, /* allowRelativeModuleNames */ true);
+            }
+
+            file.imports = imports || emptyArray;
+            
+            function collect(node: Node, allowRelativeModuleNames: boolean): void {
                 switch (node.kind) {
                     case SyntaxKind.ImportDeclaration:
                     case SyntaxKind.ImportEqualsDeclaration:
@@ -704,7 +710,9 @@ namespace ts {
                             break;
                         }
 
-                        (imports || (imports = [])).push(<LiteralExpression>moduleNameExpr);
+                        if (allowRelativeModuleNames || !isExternalModuleNameRelative((<LiteralExpression>moduleNameExpr).text)) {
+                            (imports || (imports = [])).push(<LiteralExpression>moduleNameExpr);
+                        }
                         break;
                     case SyntaxKind.ModuleDeclaration:
                         if ((<ModuleDeclaration>node).name.kind === SyntaxKind.StringLiteral && (node.flags & NodeFlags.Ambient || isDeclarationFile(file))) {
@@ -714,23 +722,15 @@ namespace ts {
                             // The StringLiteral must specify a top - level external module name.
                             // Relative external module names are not permitted
                             forEachChild((<ModuleDeclaration>node).body, node => {
-                                if (isExternalModuleImportEqualsDeclaration(node) &&
-                                    getExternalModuleImportEqualsDeclarationExpression(node).kind === SyntaxKind.StringLiteral) {
-                                    let moduleName = <LiteralExpression>getExternalModuleImportEqualsDeclarationExpression(node);
-                                    // TypeScript 1.0 spec (April 2014): 12.1.6
-                                    // An ExternalImportDeclaration in anAmbientExternalModuleDeclaration may reference other external modules 
-                                    // only through top - level external module names. Relative external module names are not permitted.
-                                    if (moduleName) {
-                                        (imports || (imports = [])).push(moduleName);
-                                    }
-                                }
+                                // TypeScript 1.0 spec (April 2014): 12.1.6
+                                // An ExternalImportDeclaration in anAmbientExternalModuleDeclaration may reference other external modules 
+                                // only through top - level external module names. Relative external module names are not permitted.
+                                collect(node, /* allowRelativeModuleNames */ false);
                             });
                         }
                         break;
                 }
             }
-
-            file.imports = imports || emptyArray;
         }
 
         function processSourceFile(fileName: string, isDefaultLib: boolean, refFile?: SourceFile, refPos?: number, refEnd?: number) {

--- a/src/compiler/utilities.ts
+++ b/src/compiler/utilities.ts
@@ -992,6 +992,12 @@ namespace ts {
         }
         return false;
     }
+    
+    export function isExternalModuleNameRelative(moduleName: string): boolean {
+        // TypeScript 1.0 spec (April 2014): 11.2.1
+        // An external module name is "relative" if the first term is "." or "..".
+        return moduleName.substr(0, 2) === "./" || moduleName.substr(0, 3) === "../" || moduleName.substr(0, 2) === ".\\" || moduleName.substr(0, 3) === "..\\";
+    }
 
     export function isInstantiatedModule(node: ModuleDeclaration, preserveConstEnums: boolean) {
         let moduleState = getModuleInstanceState(node);

--- a/tests/baselines/reference/exportsInAmbientModules1.js
+++ b/tests/baselines/reference/exportsInAmbientModules1.js
@@ -1,0 +1,13 @@
+//// [tests/cases/compiler/exportsInAmbientModules1.ts] ////
+
+//// [external.d.ts]
+
+export var x: number
+
+//// [main.ts]
+
+declare module "M" {
+    export {x} from "external"
+}
+
+//// [main.js]

--- a/tests/baselines/reference/exportsInAmbientModules1.symbols
+++ b/tests/baselines/reference/exportsInAmbientModules1.symbols
@@ -1,0 +1,11 @@
+=== tests/cases/compiler/external.d.ts ===
+
+export var x: number
+>x : Symbol(x, Decl(external.d.ts, 1, 10))
+
+=== tests/cases/compiler/main.ts ===
+
+declare module "M" {
+    export {x} from "external"
+>x : Symbol(x, Decl(main.ts, 2, 12))
+}

--- a/tests/baselines/reference/exportsInAmbientModules1.types
+++ b/tests/baselines/reference/exportsInAmbientModules1.types
@@ -1,0 +1,11 @@
+=== tests/cases/compiler/external.d.ts ===
+
+export var x: number
+>x : number
+
+=== tests/cases/compiler/main.ts ===
+
+declare module "M" {
+    export {x} from "external"
+>x : number
+}

--- a/tests/baselines/reference/exportsInAmbientModules2.js
+++ b/tests/baselines/reference/exportsInAmbientModules2.js
@@ -1,0 +1,13 @@
+//// [tests/cases/compiler/exportsInAmbientModules2.ts] ////
+
+//// [external.d.ts]
+
+export default class C {}
+
+//// [main.ts]
+
+declare module "M" {
+    export * from "external"
+}
+
+//// [main.js]

--- a/tests/baselines/reference/exportsInAmbientModules2.symbols
+++ b/tests/baselines/reference/exportsInAmbientModules2.symbols
@@ -1,0 +1,11 @@
+=== tests/cases/compiler/external.d.ts ===
+
+export default class C {}
+>C : Symbol(C, Decl(external.d.ts, 0, 0))
+
+=== tests/cases/compiler/main.ts ===
+
+No type information for this code.declare module "M" {
+No type information for this code.    export * from "external"
+No type information for this code.}
+No type information for this code.

--- a/tests/baselines/reference/exportsInAmbientModules2.types
+++ b/tests/baselines/reference/exportsInAmbientModules2.types
@@ -1,0 +1,11 @@
+=== tests/cases/compiler/external.d.ts ===
+
+export default class C {}
+>C : C
+
+=== tests/cases/compiler/main.ts ===
+
+No type information for this code.declare module "M" {
+No type information for this code.    export * from "external"
+No type information for this code.}
+No type information for this code.

--- a/tests/baselines/reference/importsInAmbientModules1.js
+++ b/tests/baselines/reference/importsInAmbientModules1.js
@@ -1,0 +1,13 @@
+//// [tests/cases/compiler/importsInAmbientModules1.ts] ////
+
+//// [external.d.ts]
+
+export var x: number
+
+//// [main.ts]
+
+declare module "M" {
+    import {x} from "external"
+}
+
+//// [main.js]

--- a/tests/baselines/reference/importsInAmbientModules1.symbols
+++ b/tests/baselines/reference/importsInAmbientModules1.symbols
@@ -1,0 +1,11 @@
+=== tests/cases/compiler/external.d.ts ===
+
+export var x: number
+>x : Symbol(x, Decl(external.d.ts, 1, 10))
+
+=== tests/cases/compiler/main.ts ===
+
+declare module "M" {
+    import {x} from "external"
+>x : Symbol(x, Decl(main.ts, 2, 12))
+}

--- a/tests/baselines/reference/importsInAmbientModules1.types
+++ b/tests/baselines/reference/importsInAmbientModules1.types
@@ -1,0 +1,11 @@
+=== tests/cases/compiler/external.d.ts ===
+
+export var x: number
+>x : number
+
+=== tests/cases/compiler/main.ts ===
+
+declare module "M" {
+    import {x} from "external"
+>x : number
+}

--- a/tests/baselines/reference/importsInAmbientModules2.js
+++ b/tests/baselines/reference/importsInAmbientModules2.js
@@ -1,0 +1,13 @@
+//// [tests/cases/compiler/importsInAmbientModules2.ts] ////
+
+//// [external.d.ts]
+
+export default class C {}
+
+//// [main.ts]
+
+declare module "M" {
+    import C from "external"
+}
+
+//// [main.js]

--- a/tests/baselines/reference/importsInAmbientModules2.symbols
+++ b/tests/baselines/reference/importsInAmbientModules2.symbols
@@ -1,0 +1,11 @@
+=== tests/cases/compiler/external.d.ts ===
+
+export default class C {}
+>C : Symbol(C, Decl(external.d.ts, 0, 0))
+
+=== tests/cases/compiler/main.ts ===
+
+declare module "M" {
+    import C from "external"
+>C : Symbol(C, Decl(main.ts, 2, 10))
+}

--- a/tests/baselines/reference/importsInAmbientModules2.types
+++ b/tests/baselines/reference/importsInAmbientModules2.types
@@ -1,0 +1,11 @@
+=== tests/cases/compiler/external.d.ts ===
+
+export default class C {}
+>C : C
+
+=== tests/cases/compiler/main.ts ===
+
+declare module "M" {
+    import C from "external"
+>C : typeof C
+}

--- a/tests/baselines/reference/importsInAmbientModules3.js
+++ b/tests/baselines/reference/importsInAmbientModules3.js
@@ -1,0 +1,13 @@
+//// [tests/cases/compiler/importsInAmbientModules3.ts] ////
+
+//// [external.d.ts]
+
+export default class C {}
+
+//// [main.ts]
+
+declare module "M" {
+    import C = require("external");
+}
+
+//// [main.js]

--- a/tests/baselines/reference/importsInAmbientModules3.symbols
+++ b/tests/baselines/reference/importsInAmbientModules3.symbols
@@ -1,0 +1,11 @@
+=== tests/cases/compiler/main.ts ===
+
+declare module "M" {
+    import C = require("external");
+>C : Symbol(C, Decl(main.ts, 1, 20))
+}
+=== tests/cases/compiler/external.d.ts ===
+
+export default class C {}
+>C : Symbol(C, Decl(external.d.ts, 0, 0))
+

--- a/tests/baselines/reference/importsInAmbientModules3.types
+++ b/tests/baselines/reference/importsInAmbientModules3.types
@@ -1,0 +1,11 @@
+=== tests/cases/compiler/main.ts ===
+
+declare module "M" {
+    import C = require("external");
+>C : typeof C
+}
+=== tests/cases/compiler/external.d.ts ===
+
+export default class C {}
+>C : C
+

--- a/tests/cases/compiler/exportsInAmbientModules1.ts
+++ b/tests/cases/compiler/exportsInAmbientModules1.ts
@@ -1,0 +1,10 @@
+// @module: amd
+
+// @filename: external.d.ts
+export var x: number
+
+// @filename: main.ts
+
+declare module "M" {
+    export {x} from "external"
+}

--- a/tests/cases/compiler/exportsInAmbientModules2.ts
+++ b/tests/cases/compiler/exportsInAmbientModules2.ts
@@ -1,0 +1,10 @@
+// @module: amd
+
+// @filename: external.d.ts
+export default class C {}
+
+// @filename: main.ts
+
+declare module "M" {
+    export * from "external"
+}

--- a/tests/cases/compiler/importsInAmbientModules1.ts
+++ b/tests/cases/compiler/importsInAmbientModules1.ts
@@ -1,0 +1,10 @@
+// @module: amd
+
+// @filename: external.d.ts
+export var x: number
+
+// @filename: main.ts
+
+declare module "M" {
+    import {x} from "external"
+}

--- a/tests/cases/compiler/importsInAmbientModules2.ts
+++ b/tests/cases/compiler/importsInAmbientModules2.ts
@@ -1,0 +1,10 @@
+// @module: amd
+
+// @filename: external.d.ts
+export default class C {}
+
+// @filename: main.ts
+
+declare module "M" {
+    import C from "external"
+}

--- a/tests/cases/compiler/importsInAmbientModules3.ts
+++ b/tests/cases/compiler/importsInAmbientModules3.ts
@@ -1,0 +1,10 @@
+// @module: amd
+
+// @filename: external.d.ts
+export default class C {}
+
+// @filename: main.ts
+
+declare module "M" {
+    import C = require("external");
+}

